YoCode/ProcessRunner.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+
+namespace YoCode
+{
+    class ProcessRunner
+    {
+        ProcessInfo procinfo;
+        public string Output { get; set; }
+
+        public ProcessRunner(string processName,string workingDir,string arguments){
+            procinfo = setupProcessInfo(processName, workingDir, arguments);
+        }
+
+        public void ExecuteTheCheck()
+        {
+            var p = new Process();
+            ExecuteTheCheck(p);
+        }
+
+        public void ExecuteTheCheck(Process p)
+        {
```

Comment:
I think this should be private because I wouldn't expect anybody to call this function and supply a `Process` object. In fact, I think I would remove it and just keep the `ExecuteTheCheck()` method with no parameters.